extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosImporter.java

```diff
@@ -202,13 +205,20 @@ PhotoResult importSinglePhoto(
     NewMediaItemUpload uploadItem =
         new NewMediaItemUpload(albumId, Collections.singletonList(newMediaItem));
     try {
-      return new PhotoResult(
-          getOrCreatePhotosInterface(jobId, authData)
-              .createPhoto(uploadItem)
-              .getResults()[0]
-              .getMediaItem()
-              .getId(),
-          bytes);
+      BatchMediaItemResponse photoCreationResponse =
+          getOrCreatePhotosInterface(jobId, authData).createPhoto(uploadItem);
+      if (photoCreationResponse == null) {
+        throw new IOException("The photo was not created");
+      }
+      NewMediaItemResult[] mediaItemResults = photoCreationResponse.getResults();
+      if (mediaItemResults == null || mediaItemResults.length != 1) {
+        throw new IOException("No media item results returned");
+      }
+      GoogleMediaItem mediaItem = mediaItemResults[0].getMediaItem();
+      if (mediaItem == null) {
+        throw new IOException("No media item returned");
+      }
+      return new PhotoResult(mediaItem.getId(), bytes);
```

Comment:
Would it be better here to use a preconditon checkNotNull?

Reply:
@seehamrun This touches on a question I have about the behaviour of the idempotentImportExecutor. The executor only swallows IOExceptions, was there a particular reason for this choice? My instinct is that this sort of issue with the API response is the sort of thing we would want the executor to handle but using precondition checkNotNull would not handle this.

Reply:
Sorry for the late response, that's a good point. I dont think we had any specific reason for this. We wanted to distinguish between framwork errors and api errors somehow and this seemed like the easiest way to do so. 

However, we can revisit this as well. Do you want to file a github issue to follow up on it?